core/src/main/java/com/linecorp/armeria/internal/common/metric/RequestMetricSupport.java

```diff
@@ -142,6 +142,7 @@ private static void onResponse(
         final int childrenSize = log.children().size();
         if (childrenSize > 0) {
             metrics.actualRequests().increment(childrenSize);
+            metrics.retries().record(childrenSize);
```

Comment:
This isn't directly related to your change, but I think we also want to remove the `return` statement at L137 since we want to record retries even if they ended up with a `WriteTimeoutException`.

https://github.com/line/armeria/blob/20af1955a62f088daed57bbbefc219bf36f6272f/core/src/main/java/com/linecorp/armeria/internal/common/metric/RequestMetricSupport.java#L135